### DISCUSSION TITLE
How to fix nuxt redirect to homepage on hard refresh？

### DISCUSSION THREAD

**Discussion by Michael0520** [nuxt/nuxt#25444] [Questions]
I am developing a web application using the Nuxt framework and deploying it on the Vercel platform. The project contains several pages, such as IndexPage, IdPage, and SearchPage.

The problem I am currently facing is that when I try to access specific paths of these pages, like SearchPage or IdPage, the system always redirects me back to IndexPage.

For example, when I attempt to access the following URLs:

IndexPage: https://final-nuxt3-app.vercel.app/
SearchPage: https://final-nuxt3-app.vercel.app/search?q=cap
AboutPage: https://final-nuxt3-app.vercel.app/about
IdPage: https://final-nuxt3-app.vercel.app/products/2
However, no matter which URL I visit, I always end up seeing only the content of the IndexPage.

But this is good work in localhost, so i can fix it in vercel?

---
Demo Page :https://final-nuxt3-app.vercel.app/
Demo Repo: https://github.com/Michael0520/final-nuxt3-app

**Comment by emamipouya** [comment]
i had same problem with my nuxt3 and Nginx, it was related to nginx setting
instead of default config like: 
location / {
    try_files $uri $uri/ /index.html;
    access_log /var/log/nginx/debug.log;
}
i used this configuration: 
location / {
    proxy_pass http://127.0.0.1:3000; # Forward to the Nuxt server
    proxy_http_version 1.1;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection 'upgrade';
    proxy_set_header Host $host;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
}
after these changes the problem resolved and pages reloaded without redirecting to homepage
